src/System.Text.Json/src/System/Text/Json/Serialization/JsonPropertyInfo.cs

```diff
@@ -353,7 +353,8 @@ public void Read(JsonTokenType tokenType, ref ReadStack state, ref Utf8JsonReade
             Debug.Assert(ShouldDeserialize);
 
             JsonPropertyInfo propertyInfo;
-            if (ElementClassInfo != null && (propertyInfo = ElementClassInfo.PolicyProperty) != null)
+            JsonClassInfo elementClassInfo = ElementClassInfo;
```

Comment:
What's the rationale behind this change?

Reply:
Presumably the getter for ElementClassInfo does some work and this avoids doing it twice?

Reply:
> Presumably the getter for ElementClassInfo does some work

Yes, of course.

> this avoids doing it twice?

I missed the second use of `elementClassInfo` (when setting `propertyInfo`). Looking more closely, makes sense now :)